From 6aacd6e5e80181dbdbf99d921b692cb98d892bf1 Mon Sep 17 00:00:00 2001
From: Noam Camus <ncamus@habana.ai>
Date: Mon, 08 Apr 2019 18:21:36 +0300
Subject: [PATCH] [SW-0] preboot: remove last unused u-boot headers

Remove headers and path search to u-boot to eliminate u-boot coupling

Finally !!!

Change-Id: I7011ac48782e21b0d403a10a7be34a50772ecfd7
---

diff --git a/Makefile b/Makefile
index dcd3cd4..9472957 100644
--- a/Makefile
+++ b/Makefile
@@ -2,9 +2,6 @@
 OBJCOPY = $(CROSS_COMPILE)objcopy
 CFLAGS += -I$(dir $(BR2_EXTERNAL))/specs/$(HL_DEV)/asic_reg
 CFLAGS += -I$(dir $(BR2_EXTERNAL))/embedded/specs
-CFLAGS += -D__KERNEL__
-CFLAGS += -I ../uboot-custom/include
-CFLAGS += -I ../uboot-custom/arch/arm/include
 CFLAGS += -Wall -Werror
 CFLAGS += -g -c
 CFLAGS += -I ./ -I ./common/
diff --git a/common/preboot.c b/common/preboot.c
index 058ce0c..dfbd364 100644
--- a/common/preboot.c
+++ b/common/preboot.c
@@ -5,10 +5,8 @@
  * SPDX-License-Identifier:     GPL-2.0+
  */
 
-#include <linux/io.h>
 #include "auto_generated.h"
 #include <stddef.h>
-#include <stdbool.h>
 
 #include <psoc_global_conf_regs.h>
 #include <hl_fw_if.h>
diff --git a/common/spi.c b/common/spi.c
index bdaed34..650d3c7 100644
--- a/common/spi.c
+++ b/common/spi.c
@@ -5,7 +5,7 @@
  * SPDX-License-Identifier:     GPL-2.0+
  */
 
-#include <linux/io.h>
+#include <stddef.h>
 
 #include <psoc_gpio3_regs.h>
 #include <psoc_spi_regs.h>
diff --git a/gaudi/gaudi.c b/gaudi/gaudi.c
index 28cf7bf..a401588 100644
--- a/gaudi/gaudi.c
+++ b/gaudi/gaudi.c
@@ -5,8 +5,6 @@
  * SPDX-License-Identifier:     GPL-2.0+
  */
 
-#include <linux/io.h>
-
 #include <psoc_global_conf_regs.h>
 #include <pcie_aux_regs.h>
 
diff --git a/goya/goya.c b/goya/goya.c
index db60bab..d8b889b 100644
--- a/goya/goya.c
+++ b/goya/goya.c
@@ -5,8 +5,6 @@
  * SPDX-License-Identifier:     GPL-2.0+
  */
 
-#include <linux/io.h>
-
 #include <pcie_phy_regs.h>
 #include <psoc_global_conf_regs.h>
 #include <pcie_aux_regs.h>
diff --git a/goya/pll.c b/goya/pll.c
index 817934b..cf0af4d 100644
--- a/goya/pll.c
+++ b/goya/pll.c
@@ -1,5 +1,3 @@
-#include <linux/io.h>
-
 #include "preboot.h"
 #include "uart.h"
 
